File: app/models/asset_category.py
from typing import TYPE_CHECKING

from sqlalchemy import String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base

if TYPE_CHECKING:
    from app.models.asset import Asset


class AssetCategory(Base):
    __tablename__ = "asset_categories"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    nome: Mapped[str] = mapped_column(String(100), unique=True, nullable=False)
    descricao: Mapped[str | None] = mapped_column(Text, nullable=True)

    assets: Mapped[list["Asset"]] = relationship("Asset", back_populates="categoria")

File: backend/internal/models/asset_category.go
package models

type AssetCategory struct {
	ID        uint    `gorm:"primaryKey" json:"id"`
	Nome      string  `gorm:"uniqueIndex;not null;size:100" json:"nome"`
	Descricao *string `gorm:"type:text" json:"descricao"`
}

func (AssetCategory) TableName() string { return "asset_categories" }
